server/registry/internal/storage/filtering/filtering.go

```diff
@@ -60,22 +55,22 @@ func (f *Filter) Matches(model map[string]interface{}) (bool, error) {
 	return match, nil
 }
 
-func NewFilter(filter string, fields []Field) (Filter, error) {
+func NewFilter(filter string, fields map[string]FieldType) (Filter, error) {
```

Comment:
Is this change related to ordering support? (I'm not seeing a relationship on initial reading)

Reply:
Good catch, I made this change at an intermediate step during development because I wanted to avoid referencing types from the `filtering` package. That didn't work out in the end: `func gormOrdering(ordering string, fields map[string]filtering.FieldType)` was the best I could do instead of `func gormOrdering(ordering string, fields []filtering.Field)`.

That said, this change isn't really important to the PR (although I think it's still worth making - this type really doesn't add value). I can take some time to revert this and open a new PR with the small improvement, but IMO it's insignificant enough to keep it here and save the effort of reorganizing. If it wasn't lumped into a larger commit I could easily separate it, but that's not the case here.

Let me know if you want me to separate it or just drop it entirely, otherwise I'll leave it here.

Reply:
Thanks, to keep our commits easy to review I'd like to do this in two PRs - a first one that makes the relatively low impact / high line count change to the filters and then we can merge that into this one which will hopefully then be a high impact / low line count change that is easy to refer back to.

Reply:
Regarding the filtering change specifically: I agree that the exported type isn't adding any value. The iteration order becomes unstable when we use a map, but that shouldn't be affecting anything (considering the way this is used).

Reply:
#678